service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +74,279 @@ public void disbandTeam(JWTData userData, int teamId) {
     team.setDeletedAt(new Timestamp(System.currentTimeMillis()));
     team.store();
   }
+
+  @Override
+  public void createTeam(JWTData userData, CreateTeamRequest request) {
+    TeamsRecord team = db.newRecord(TEAMS);
+    team.setBio(request.getBio());
+    team.setTeamName(request.getName());
+    team.store();
+    UsersTeamsRecord usersTeam = db.newRecord(USERS_TEAMS);
+    usersTeam.setTeamId(team.getId());
+    usersTeam.setTeamRole(TeamRole.LEADER);
+    usersTeam.setUserId(userData.getUserId());
+    usersTeam.store();
+  }
+
+  @Override
+  public TeamDataResponse getTeam(JWTData userData, int teamId) {
+    TeamsRecord team = db.selectFrom(TEAMS).where(TEAMS.ID.eq(teamId)).fetchOne();
+    if (team == null) {
+      throw new ResourceDoesNotExistException(teamId, "Team");
+    }
+
+    return new TeamDataResponse(
+        team.getId(),
+        team.getTeamName(),
+        team.getBio(),
+        team.getFinished(),
+        team.getCreatedAt(),
+        team.getDeletedAt());
+  }
+
+  @Override
+  public void addGoal(JWTData userData, AddGoalRequest addGoalRequest, int teamId) {
+    checkTeamExists(teamId);
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      GoalsRecord goal = db.newRecord(GOALS);
+      goal.setTeamId(teamId);
+      goal.setGoal(addGoalRequest.getGoal());
+      goal.setStartAt(addGoalRequest.getStart_at());
+      goal.setCompleteBy(addGoalRequest.getComplete_by());
+      goal.store();
+    } else {
+      throw new WrongTeamRoleException(teamId, usersTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void deleteGoal(JWTData userData, int teamId, int goalId) {
+    checkTeamExists(teamId);
+    checkGoalExists(goalId);
+
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      db.deleteFrom(GOALS).where(GOALS.ID.eq(goalId));
```

Comment:
I think deleting goals is better. Don't see why someone would want to restore a goal rather than just recreate it. We already have the ability to mark as complete, don't see why completed and deleted states are both necessary.

Reply:
Sounds good 👍